frontend/src/components/parking-lots-list/parking-lot-state-toggle-button.tsx

```diff
@@ -0,0 +1,52 @@
+import { ParkingLotModel, ParkingLotState } from '../../models';
+import { ToggleButton, ToggleButtonGroup, Typography } from '@mui/material';
+import React from 'react';
+
+interface parkingLotStateToggleButtonProps {
+  parkingLot: ParkingLotModel;
+  changeParkingLotState: (item: ParkingLotModel) => void;
+}
+
+const ParkingLotStateToggleButton = ({
+  parkingLot,
+  changeParkingLotState
+}: parkingLotStateToggleButtonProps): JSX.Element => {
+  return (
+    <ToggleButtonGroup
+      value={parkingLot.state}
+      exclusive
+      onChange={() => changeParkingLotState(parkingLot)}
+    >
+      <ToggleButton
+        value={ParkingLotState.released}
+        color={'success'}
+        disabled={parkingLot.state == ParkingLotState.released}
+        style={{ minWidth: 150 }}
+      >
+        {releasedText(parkingLot.state)}
+      </ToggleButton>
+      <ToggleButton
+        value={ParkingLotState.locked}
+        color={'error'}
+        disabled={parkingLot.state == ParkingLotState.locked}
+        style={{ minWidth: 150 }}
+      >
+        {lockedText(parkingLot.state)}
+      </ToggleButton>
+    </ToggleButtonGroup>
+  );
+};
+
+const lockedText = (state: ParkingLotState) =>
+  state === ParkingLotState.locked ? (
+    <Typography>gesperrt</Typography>
+  ) : (
+    <Typography>sperren</Typography>
+  );
+const releasedText = (state: ParkingLotState) =>
+  state === ParkingLotState.released ? (
+    <Typography>freigegeben</Typography>
+  ) : (
+    <Typography>freigeben</Typography>
+  );
```

Comment:
You could put the Typography in the parent and just pass the string, saves you some extra wrappers